src/selectors/entities/channels.ts

```diff
@@ -765,6 +765,56 @@ export const getSortedUnreadChannelIds: (state: GlobalState, lastUnreadChannel:
     (unreadChannelIds, mappedAndSortedUnreadChannelIds) => mappedAndSortedUnreadChannelIds,
 );
 
+//recent channels
+
+export const getAllRecentChannels: (state: GlobalState) => Array<Channel> = createSelector(
+    getUsers,
+    getCurrentUser,
+    getAllChannels,
+    getUserIdsInChannels,
+    getLastPostPerChannel,
+    getMyChannelMemberships,
+    getChannelIdsForCurrentTeam,
+    getTeammateNameDisplaySetting,
+    (profiles, currentUser: UserProfile, channels: IDMappedObjects<Channel>, userIdsInChannels: any,
+        lastPosts: RelationOneToOne<Channel, Post>,
+        members: RelationOneToOne<Channel, ChannelMembership>,
+        teamChannelIds: Array<string>,
+        settings,
+    ): Array<Channel> => {
+        const sorting = 'recent';
+        const recentIds = teamChannelIds.filter((id) => {
+            const c = channels[id];
+            const m = members[id];
+
+            if (c && m) {
+                return true;
+            }
+
+            return false;
```

Comment:
Also, this can be simplified slightly to

```suggestion
            return Boolean(c && m);
```